### PR TITLE
feat: Add Legacy Report View

### PR DESCRIPTION
**Pull request by dhaselhan**
* Add new brancher to control legacy vs non-legacy view
* Split HistoryCard to its own component for re-usability and add tests
* Update legacy view with legacy assessment card
* Sent legacy_id to the front end

### CODE REVIEW

File: frontend/public/config/config.js [hunk 1]
```diff
@@ -1,5 +1,6 @@
 export const config = {
   api_base: 'http://localhost:8000/api',
+  tfrs_base: 'http://localhost:3001',
```
Comment (AlexZorkin) [inline frontend/public/config/config.js:3]: in local we're using port 3000

File: frontend/src/App.jsx [hunk 2]
```diff
@@ -196,7 +198,7 @@ const router = createBrowserRouter([
       },
       {
         path: ROUTES.REPORTS_VIEW,
-        element: <EditViewComplianceReport />,
+        element: <ViewComplianceReportBrancher />,
```
Comment (AlexZorkin) [inline frontend/src/App.jsx:201]: maybe a different name 'ComplianceReportViewSelector'?

**Review by AlexZorkin — CHANGES_REQUESTED** [review]
Looks great, only 2 comments.

File: frontend/public/config/config.js (hunk 1 above)
Comment (dhaselhan) [reply frontend/public/config/config.js:3]: I set this to 3001 as LCFS also runs on 3000 so if you were to run both locally at the same time TFRS would need to be on something else. Should I set them to the same?

File: frontend/src/App.jsx (hunk 2 above)
Comment (dhaselhan) [reply frontend/src/App.jsx]: Done!

File: frontend/public/config/config.js (hunk 1 above)
Comment (AlexZorkin) [reply frontend/public/config/config.js:3]: I do like the idea but there are quite a few places we would have to update this. vite config, cypress config, application settings, and a few more spots. Maybe leave it as 3000 and then adjust locally when testing this hybrid code. If you want to go through and update it everywhere im not opposed but we would have to test thoroughly.

Comment (dhaselhan) [reply frontend/public/config/config.js]: Updated to 3000.

**Review by AlexZorkin — APPROVED** [review]
👍

**Pull request merged**